File: src/CSharp/102.BinaryTreeLevelOrderTraversal.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;


namespace PeefyLeetCode.BinaryTreeLevelOrderTraversal
{

public class TreeNode {
    public int val;
    public TreeNode left;
    public TreeNode right;
    public TreeNode(int x) { val = x; }
}

public class Solution {
    public IList<IList<int>> LevelOrder(TreeNode root) {
        IList<IList<int>> ans = new List<IList<int>>();
        if (root == null)
            return ans;
        List<TreeNode> level = new List<TreeNode>();
        level.Add(root);
        while (level.Count > 0){
            List<int> cur = new List<int>();
            List<TreeNode> next_level = new List<TreeNode>();
            foreach (TreeNode node in level) {
                cur.Add(node.val);
                if (node.left != null)
                    next_level.Add(node.left);
                if (node.right != null)
                    next_level.Add(node.right);
            }
            ans.Add(cur);
            level = next_level;
        }
        return ans;
    }
}
}

File: src/Python/102.BinaryTreeLevelOrderTraversal.py

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def levelOrder(self, root: TreeNode) -> list:
        if root is None:
            return []
        ans = []
        tmp = []
        queue = [root]
        nextqueue = []
        while len(queue) > 0:
            node = queue.pop(0)
            tmp.append(node.val)
            if node.left is not None:
                nextqueue.append(node.left)
            if node.right is not None:
                nextqueue.append(node.right)
            if len(queue) == 0:
                ans.append(tmp[:])
                tmp.clear()
                queue = nextqueue[:]
                nextqueue.clear()
        return ans

if __name__ == "__main__":
    solution = Solution()
    node = TreeNode(2)
    node.left = TreeNode(1)
    node.right = TreeNode(3)
    node.right.left = TreeNode(4)
    print(solution.levelOrder(node))
   